pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/nonpepperkit/NonPepperKitCreationService.java

```diff
@@ -0,0 +1,276 @@
+        //        orderKits.add(kit);
+
+        return transactionResults;
+    }
+
+    private SimpleResult addJuniperKitRequest(Connection conn, String kitTypeName, KitRequestSettings kitRequestSettings,
+                                              DDPInstance ddpInstance,
+                                              int kitTypeId, String collaboratorParticipantId, String errorMessage,
+                                              EasyPostUtil easyPostUtil,
+                                              JuniperKitRequest kit, String externalOrderNumber, String juniperKitRequestId,
+                                              String ddpLabel, String userId, SimpleResult transactionResults) {
+        String collaboratorSampleId = null;
+        String bspCollaboratorSampleType = kitTypeName;
+        String addressId = null;
+        try {
+            Address address = easyPostUtil.getAddress(kit.getEasypostAddressId());
+            if (address != null) {
+                addressId = address.getId();
+            }
+        } catch (EasyPostException e) {
+            throw new RuntimeException("EasyPost addressId could not be received ", e);
+        }
+
+        if (StringUtils.isNotBlank(kitRequestSettings.getExternalShipper())) {
+            try {
+                // collaborator sample id is the id of the physical sample that GP will work with.
+                collaboratorSampleId =
+                        KitRequestShipping.generateBspSampleID(conn, collaboratorParticipantId, bspCollaboratorSampleType, kitTypeId);
+                KitRequestShipping.writeRequest(ddpInstance.getDdpInstanceId(), juniperKitRequestId, kitTypeId,
+                        kit.getParticipantId().trim(),
+                        collaboratorParticipantId, collaboratorSampleId, userId, addressId, errorMessage, externalOrderNumber, false,
+                        null, ddpInstance, bspCollaboratorSampleType, ddpLabel);
+                //             Not needed now
+                //            kit.setExternalOrderNumber(externalOrderNumber);
+            } catch (Exception e) {
+                transactionResults.resultException = e;
+            }
+        } else {
+            if (kitRequestSettings.getCollaboratorSampleTypeOverwrite() != null) {
+                bspCollaboratorSampleType = kitRequestSettings.getCollaboratorSampleTypeOverwrite();
+            }
+            if (StringUtils.isNotBlank(collaboratorParticipantId)) {
+                collaboratorSampleId =
+                        KitRequestShipping.generateBspSampleID(conn, collaboratorParticipantId, bspCollaboratorSampleType, kitTypeId);
+                if (collaboratorParticipantId == null) {
+                    errorMessage += "collaboratorParticipantId was too long ";
+                }
+                if (collaboratorSampleId == null) {
+                    errorMessage += "collaboratorSampleId was too long ";
```

Comment:
I don't think this works (this will not update errorMessage for the caller). Do you have a test for this?
Also, it seems like you are attempting to return errors in two different ways, via the return and via this parameter. One way would be less confusing.

Reply:
No the error message here will be written to the database. It is one of the GP errors if we can't fit a participant id into their desired length for barcodes

Reply:
Oh, ok. So this is a non-fatal error? Does GP view it? If so, how?

I'm unclear on why errorMessage is passed into addJuniperKitRequest. It appears to be empty when passed.